shorturl/views.py

```diff
@@ -79,4 +80,40 @@ def is_already_short(url):
     for shortening_url in popular_shortening_urls:
         if shortening_url in url:
             return True
-    return False
\ No newline at end of file
+    return False
+
+def validate_url(url):
+	protocols=[
+		'http://',
+		'https://'
+	]
+	flag=0
+	url_form_field = URLField()
+	email_field = EmailField()
+	try :
+		email_field.clean(url)
+	except ValidationError:
+		if url!='':
+			for protocol in protocols:
+				n=len(protocol)
+				if url[0:n]==protocol:
+					flag=1
+					break
+			if flag==0:
+				flag1=1
+				for protocol in protocols:
+					new_url = protocol+url
+					try:
+						new_url == url_form_field.clean(new_url)
+					except ValidationError:
+						flag1=0
+					else:
+						url=new_url
+						break
+				if flag1==1:
+					return True,url
+				return False,url
+			return True,url
+		return False,url
+	else:
```

Comment:
Is this else part of any if-else block?

Reply:
Okay got it. But is it really required out there? Okay let it be for now. I will see it later myself.

Reply:
Yes it is indeed required because I am trying whether the url is valid then turning towards the exception clause and if it does not give an exception then it should return the proper output i cannot return the output from the try clause.